### DISCUSSION TITLE
phpunit testable livewire fails with Undefined array key "fingerprint"

### DISCUSSION THREAD

**Discussion by kieferjs** [livewire/livewire#5537] [1. Help] [answered]
When trying a test that came with Laravel and Jetstream/Livewire libraries, I get an undefined array key "fingerprint" error message

Undefined array key "fingerprint"

at vendor/livewire/livewire/src/Testing/TestableLivewire.php:181
    public function pretendWereSendingAComponentUpdateRequest($message, $payload)
  {
       $result = $this->callEndpoint('POST', '/livewire/message/'.$this->componentName, [
           'fingerprint' => $this->payload['fingerprint'],
           'serverMemo' => $this->payload['serverMemo'],
           'updates' => [['type' => $message, 'payload' => $payload]],
        ]);

This happens for any out of the box feature tests that ship with Laravel9 with Jetstream when used against my project.

Here is one example that fails at the Livewire::test.... line. The user is created and authenticating without issue and confirmed in other phpunit tests.

    public function test_other_browser_sessions_can_be_logged_out(): void
    {

        $this->actingAs($user = User::factory()->create());

        Livewire::test(LogoutOtherBrowserSessionsForm::class)
                ->set('password', $user->password)
                ->call('logoutOtherBrowserSessions')
                ->assertSuccessful();
    }

I stood up a fresh Laravel 9 project which works and began inserting various areas from my project into the fresh project as a way of hopefully identifying the issue. Session parameters, events, migrations, factories, models, were not the issue as it continued to work in the fresh project.

One thing I noticed is that the generic routes are not accepted in my project within the test cases. I have to insert 'https://realtor.host' in front of every test route (e.g. $response = $this->get('https://realtor.host/register');

I was curious if it was not evaluating the livewire route and I tried to add my domain into the vendor's livewire component in which the test still failed and that did not cause it to work.

Any ideas on where else I can look?

**Comment by kieferjs** [comment, marked as answer]
I resolved my issue.

 I had to do with something wrong with the url setup in the app.php file that only affected the test cases.

Below is the newly commented out line and replaced line below it.

//    'url' => env('APP_URL') || 'default',
    'url' => env('APP_URL'),

**Reply by joshhanley** [reply]
@kieferjs thanks for sharing your fix! 🙂
